### ISSUE TITLE
Mongodb introspection fails with "called `Option::unwrap()` on a `None` value"

### ISSUE THREAD

**Issue by alavkx** [prisma/prisma#14135]
Hi Prisma Team! Prisma Migrate just crashed. 

## Command

`db pull`

## Versions
      
| Name        | Version            |
|-------------|--------------------|
| Platform    | darwin-arm64       | 
| Node        | v16.15.0           | 
| Prisma CLI  | 4.0.0              | 
| Engine      | da41d2bb3406da22087b849f0e911199ba4fbf11| 

## Error
```
Error: [introspection-engine/connectors/mongodb-introspection-connector/src/sampler/statistics/indices.rs:222:67] called `Option::unwrap()` on a `None` value
```

**Comment by pimeys** [comment]
Do you have an index such as:

```json
{ "a.b.c": 1 }
```

defined in your database, but then in your data `b` is not an embedded type but a scalar? We cannot decipher any working data model for this, but at least we should not crash.

**Comment by alavkx** [comment]
I'm brand new to this codebase, and to mongo. Do you know a way I grab a list of all the indexes in the db? I wrote a script last week to get a list of keys for every collection, but I _think_ indexes are something different..?

**Comment by pimeys** [comment]
So you should check in the data if any of the following fields points to an embedded type in _every single document_:

- `venues.default`
- `orderstats.totalsByAgeGroup`
- `orderstats.totalsByAgeGroup.ageGroup6`
- `venuerewards.alpineIQ`
- `products.leaflogix`
- `venueoffers.springbig`
- `ordercarts.discoungs`
- `ordercarts.discounts.alpineIQ`
- `inventories.leaflogix`
- `reports.default`

I'm pushing a fix that does not crash here anymore, but we cannot really generate an immediately working data model if the data is super dirty... Of course it's easier to fix if you at least get some data model out...

**Comment by alavkx** [comment]
Is there a straightforward way to point npx to a dev branch? I can run again with your change-set and let you know the result.

**Comment by pimeys** [comment]
Yeah when a dev is out. The PR is still not approved by anybody and I'm not sure is this the solution we choose. I'll mark the dev release here when one is out.

**Comment by alavkx** [comment]
Ah saw the merge but didn't know how to tie that to a version number. Cheers I'll try that now.

**Comment by alavkx** [comment]
`yarn add -D prisma@4.1.0-dev.40 && yarn prisma db pull` worked!!! U are the mans ᕙ༼ຈل͜ຈ༽ᕗ thank you so much ❤️ ‼️ 
![](https://c.tenor.com/QH2EWyo9l4UAAAAC/lfg-lets-fucking-go.gif)

**Comment by pimeys** [comment]
Did the resulting data model make any sense?

**Comment by alavkx** [comment]
We have a giant data model, 4300 lines. I'd need someone more experienced with the model to review it all, but it generally looks really solid! I'm working through caught issues with the model to get `prisma generate` working 👍 
The errors types I've seen so far
- Field "x" is already defined on composite type "y". 
- Error validating model "x": The index definition refers to the unknown fields: ys.
